Added mentions email notifications

ronaldlangeveld opened the pull request:
refs https://github.com/TryGhost/Team/issues/2429

- started created a new MentionsNotifications class that sends an email when a new webmention is received.
- is triggered by a an event listener

ronaldlangeveld committed:
Added mentions email notifications

refs https://github.com/TryGhost/Team/issues/2429

- started created a new MentionsNotifications class that sends an email
  when a new webmention is received.
- still need to trigger the notification via an event

ronaldlangeveld committed:
Added Mention event listener

ref https://github.com/TryGhost/Team/issues/2429

- added event listener to email mention notifications
- still need to add testing

ronaldlangeveld committed:
Cleanup

ronaldlangeveld committed:
Added webmention email flag to listener

- ensure the listener only fires when the feature flag is enabled.

ronaldlangeveld committed:
Added e2e test for receiving emails

refs https://github.com/TryGhost/Team/issues/2429

- added e2e test for receiving webmention notifications
- fixed the condition in the flag inside the event listener.

ronaldlangeveld committed:
Added unit test for MentionNotifications

refs https://github.com/TryGhost/Team/issues/2429

- added testing for MentionNotifications inside the webmentions package
- removed unused code in MentionNotifications

ronaldlangeveld committed:
Merge branch 'main' into mentions-email

ronaldlangeveld committed:
Fixed test

- test broke after merging with main

allouis commented on ghost/webmentions/lib/MentionsAPI.js:
```diff
@@ -160,8 +162,9 @@ module.exports = class MentionsAPI {
                 sourceFavicon: metadata.favicon,
                 sourceFeaturedImage: metadata.image
             });
+            const event = MentionCreatedEvent.create({data: mention, timestamp: new Date()});
```
Can we update this to use the pattern we use in Tiers?

First we add the Event to the Tier object like this: https://github.com/TryGhost/Ghost/blob/main/ghost/tiers/lib/Tier.js#L282-L284

And then in the TierRepository, we dispatch all necessary events: https://github.com/TryGhost/Ghost/blob/main/ghost/core/core/server/services/tiers/TierRepository.js#L112-L114

allouis reviewed (commented):
Looks good! One comment on the event stuff :) 

Only other thing was discussed in Slack - I think it's better to use the StaffService here, rather than create a new module for it

ronaldlangeveld committed:
Added Mentions Notifications as a staff service

refs https://ghost.slack.com/archives/C02G9E68C/p1674626598287329?thread_ts=1674617402.038549&cid=C02G9E68C

- added mentions notifications to staff services and is now handled
  better via the Domain Events
- Added tests

ronaldlangeveld committed:
Cleaned up blank space

allouis reviewed (commented):
This looks great! Just one comment about where we do the labs check - but otherwise 👍

ronaldlangeveld committed:
Swapped out labs flag

refs https://github.com/TryGhost/Ghost/pull/16170/files/d80f7a7743c0a1a57d07fb3a9256f294974e9995#r1086427941

- swaped out labs flag to block the sending of the notification instead
  of the whole event.

ronaldlangeveld committed:
Merge branch 'main' into mentions-email

ronaldlangeveld committed:
Fixed unit test

- missing labs mock in e2e test
